### DISCUSSION TITLE
Getting memory size of a DuckDB database

### DISCUSSION THREAD

**Discussion by rkennedy-mode** [duckdb/duckdb#2914] [Q&A] [answered]
Is there a way to determine the (even approximate) size of a DuckDB instance's memory use? I see there's an Allocator object hanging off of the DBConfig, which I suppose means we could pass a custom allocator and implement the book keeping ourselves.

I was hoping maybe there was already something built-in to DuckDB that we could utilize, though.

**Comment by Mytherin** [comment, marked as answer]
`PRAGMA database_size` returns the current memory usage by the buffer manager. Note that this does not include all memory usage (e.g. memory usage of indexes is not currently counted by the buffer manager, neither is memory usage of small intermediates), but it should provide a good estimate in most cases.

**Reply by rkennedy-mode** [reply]
Ah, not perfect but it'll probably do for now. Thanks!

It doesn't appear to be documented with the other [pragmas](https://duckdb.org/docs/sql/pragmas). Is that just an artifact of #2097?

**Reply by l1t1** [reply]
it is shown in my Android termux. seems different from result of the free command. 
```
D PRAGMA database_size
> ;
┌───────────────┬────────────┬──────────────┬─────────────┬─────────────┬──────────┬──────────────┬──────────────┐
│ database_size │ block_size │ total_blocks │ used_blocks │ free_blocks │ wal_size │ memory_usage │ memory_limit │
├───────────────┼────────────┼──────────────┼─────────────┼─────────────┼──────────┼──────────────┼──────────────┤
│               │            │              │             │             │          │ 0 bytes      │ 6.4GB        │
└───────────────┴────────────┴──────────────┴─────────────┴─────────────┴──────────┴──────────────┴──────────────┘
D .exit
~ $ free
               total        used        free      shared  buff/cache   available
Mem:         7923772     5536244      391340       67188     1996188     1977312
Swap:        4194300     4018056      176244
```

**Comment by Alex-Monahan** [comment]
We haven't automated that just yet, you are correct! In the meantime, would you be open to adding a PR to our [docs repo](https://github.com/duckdb/duckdb-web)? Thanks!

**Reply by rkennedy-mode** [reply]
Sure, I'll give it a shot!

**Reply by rkennedy-mode** [reply]
https://github.com/duckdb/duckdb-web/pull/151
